### PR TITLE
[ci] add an allgood job for easy [required] builds

### PR DESCRIPTION
**Pull request by graingert**
see https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert

### CODE REVIEW

File: .github/workflows/main.yml
Comment (graingert) [inline .github/workflows/main.yml:165]: looks like `steps` are required

```suggestion
  allgood:
    runs-on: ubuntu-latest
    needs: [test]
    steps:
    - name: echo allgood
      run: echo "allgood"
```

**Comment by graingert** [comment]
@jameslamb you should be able to add "required" to  `GitHub Actions / allgood (pull_request) Successful in 1s `

File: .github/workflows/main.yml
Comment (StrikerRUS) [inline .github/workflows/main.yml:168]: Could you please make wording more intelligent please? 😄

**Review by StrikerRUS — COMMENTED** [review]
@graingert Thanks a lot! Indeed, this looks like something we were looking for!
Now we need just to mark this new job as `required`, right?

image

File: .github/workflows/main.yml
Comment (graingert) [reply .github/workflows/main.yml:168]:
```suggestion
    - name: echo more intelligent please
      run: echo "more intelligent please"
```

Comment (jameslamb) [reply .github/workflows/main.yml:168]: sorry, we mean could you make it something like

```yaml
- name: simple-task
   echo: "this is a minimal task"
```

**Comment by jameslamb** [comment]
@StrikerRUS yeah, I think this is right: https://github.com/microsoft/LightGBM/pull/3351#pullrequestreview-482579697

File: .github/workflows/main.yml
Comment (jameslamb) [reply .github/workflows/main.yml:168]: and instead of `allgood`, could you please name this job `all-github-actions-tests`, to make it clearer what it is there for?

and can you add a comment linking to our discussion? https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert

So that in the future, when someone looks at this file, they understand why we have this extra job that seems to only run a small `echo`

Comment (graingert) [inline .github/workflows/main.yml:168]: I don't like `simple-task` as it's not very explanatory:

how about:

```suggestion
  all-successful:
    runs-on: ubuntu-latest
    needs: [test]
    steps:
    - name: "Proclaim that all jobs in the workflow are successful, for easy [requires]"
      run: echo "🎉"
```

Comment (jameslamb) [reply .github/workflows/main.yml:168]: echoing an emoji is ok with me! I think this test name is too long. Can you please just call is `check all tests succeeded`, and then add a comment with the discussion link like I recommended in https://github.com/microsoft/LightGBM/pull/3351#discussion_r483687906?

Comment (graingert) [reply .github/workflows/main.yml:168]: all-github-actions-tests isn't good because it's already in the "Github Actions" workflow namespace

check isn't quite accurate either, so I think commemorate is more correct:

```suggestion
  all-successful:  # https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert
    runs-on: ubuntu-latest
    needs: [test]
    steps:
    - name: commemorate that all tests succeeded
      run: echo "🎉"
```

Comment (jameslamb) [reply .github/workflows/main.yml:168]: We have many maintainers and contributors on this project who are not native English speakers, so I have a strong preference for plain language, avoiding words like `proclaim` or `commemorate`.

Please call the step `check all tests succeeded` and please put the comment on its own line (not inline with `all-successful: `).

File: .github/workflows/main.yml
Comment (graingert) [inline .github/workflows/main.yml:168]:
```suggestion
    - name: note that all tests succeeded
```

**Review by StrikerRUS — APPROVED** [review]
@graingert Thanks a lot for the workaround!

And thanks for making wordings easier to understand, LGTM!

**Review by jameslamb — APPROVED** [review]
works for me, thanks!

@StrikerRUS @guolinke I'll switch the set of required checks after merging

**Comment by jameslamb** [comment]
oh nevermind, it looks like I don't have access to change the required checks.

@guolinke sorry to bother you, could you change the set of required checks for GitHub Actions? Now we only need to require the `all-successful` task and all others can be disabled. After that, we'll be able to add and remove jobs without needing you too change the settings again.

**Comment by guolinke** [comment]
@jameslamb no problem

**Comment by graingert** [comment]
Have a look at https://github.com/probot/settings it allows setting
settings with PRs

**Comment by jameslamb** [comment]
thanks @graingert . We change these settings rarely enough that I think it's better to have them done manually than to have another integration to maintain.

File: .github/workflows/main.yml
Comment (webknjaz) [inline .github/workflows/main.yml:163]: @graingert FYI this check may be problematic with failures and branch protection. Here's a more comprehensive solution: https://github.com/re-actors/alls-green.

Comment (graingert) [reply .github/workflows/main.yml:163]: Ah nice one!

Comment (danepowell) [reply .github/workflows/main.yml:163]: Could you be more specific? How might this cause problems with branch protection?

Comment (webknjaz) [reply .github/workflows/main.yml:163]: @danepowell it's explained at https://github.com/marketplace/actions/alls-green#why but TL;DR — when some job dependencies fail, this job may end up being skipped and branch protection treats skipped jobs as success.

Comment (webknjaz) [reply .github/workflows/main.yml:163]: I've also filed https://github.com/microsoft/LightGBM/issues/5501.

**Pull request merged**
